src/components/duet-date-picker/date-utils.ts

```diff
@@ -209,3 +210,21 @@ export function chr4() {
 export function createIdentifier(prefix) {
   return `${prefix}-${chr4()}${chr4()}-${chr4()}-${chr4()}-${chr4()}-${chr4()}${chr4()}${chr4()}`
 }
+
+export function weekOfYear(date: Date): number {
+  const givenYear = date.getFullYear()
+  const fourthOfJanuaryOfGivenYear = startOfWeek(new Date(givenYear, 0, 4))
+  const fourthOfJanuaryOfNextYear = startOfWeek(new Date(givenYear + 1, 0, 4))
+  let startYear
+  if (date.getTime() >= fourthOfJanuaryOfNextYear.getTime()) {
+    startYear = givenYear + 1
+  } else if (date.getTime() >= fourthOfJanuaryOfGivenYear.getTime()) {
+    startYear = givenYear
+  } else {
+    startYear = givenYear - 1
+  }
+  const start = startOfWeek(new Date(startYear, 0, 4))
+  const end = startOfWeek(date)
+  const diff = end.getTime() - start.getTime()
+  return Math.round(diff / MILLISECONDS_IN_WEEK) + 1
```

Comment:
I'm curious about the implementation of this. It's very hard to know if this gives correct results for every possibility. Is this lifted from somewhere (which is fine but a link in a comment would be useful), or have you written it yourself? Are you able to give a walk through of what's going on here? 

I'm curious about the use of 4th January. And also whether this would be affected by (and therefore should take as an argument) the first day of the week? I assume so

Reply:
Here are the rules of how to determine week numbers https://en.wikipedia.org/wiki/Week#Week_numbering
This version returns ISO week number and is healivy inspired by momentjs and date-fns packages. Both of them use 4th january to find first week.

In my use-case iso week number made most sense as it matches values that PostgreSQL is returning(`SELECT EXTRACT('isoyear' FROM '2021-01-03'::date), EXTRACT('week' FROM '2021-01-03'::date)`) but there are other calendar apps with different logic. For example Apple Calendar 2020-12-28 is recognised as 1st week and not 53.

date-fns package getWeek function has two options - weekStartsOn and firstWeekContainsDate so solve this.

How do you feel about adding momentjs or date-fns as a dependency? Or should we just reimplement the function here as well?

Reply:
@WickyNilliams Have you had time to decide how this could be continued?

Reply:
Hey sorry about the delay @krists. Have a lot going on at the moment, so it's hard to find time to dedicate to this. 

Oh that wikipedia article will be very useful. I started researching how to calculate this before and wasn't finding any good information.

I don't want to add date-fns or moment as a dependency, since we were very focused on making this picker as lightweight as possible.

I'll take a closer look at the logic here later in the week

Reply:
@WickyNilliams understandable. jquery-ui library [has similar functionality](https://github.com/jquery/jquery-ui/blob/817ce38555f07981f929fb4b1229fc42574cf85c/ui/widgets/datepicker.js#L1105-L1120) as well. It provides a [option](https://api.jqueryui.com/datepicker/#option-calculateWeek) to override default implemenation.